### PR TITLE
Minor typing and other cleanups extracted from #3795

### PR DESCRIPTION
**Pull request by erights**
Minor typing and other cleanups extracted from #3795
Time to factor these out so they're not a distraction but don't get lost.

### CODE REVIEW

File: packages/marshal/src/helpers/internal-types.js
Comment (michaelfig) [inline packages/marshal/src/helpers/internal-types.js]: This `import` is the reason your "Why do I need these?" explicit imports are necessary.  Any JS in a JSDoc types file requires you to explicitly use `import(./internal-types.js).Foo` in the other JS files.  Without any JS statements, the types becomes ambient.

When we move from ambient types to explicit typing, you will need those explicit imports, but until we bite that bullet, you can just remove this line and the explicit imports.

File: packages/marshal/src/helpers/remotable.js
Comment (michaelfig) [inline packages/marshal/src/helpers/remotable.js]: Because your `./internal-types.js` file contains JS statements.

File: packages/vats/src/ibc.js
Comment (michaelfig) [inline packages/vats/src/ibc.js:30]: I suspect you were too aggressive in removing the definitions of several of these `PromiseRecord<T>` types.  Please search the file for occurrences of `PromiseRecord` (or `ERef`), because if you delete the explicit import, then these types default to `any`, which is a fate worse than death.

Okay, maybe not **that** bad, but still bad. :)

**Review by michaelfig — APPROVED** [review]
Approved, but please don't merge unless you have verified that the removed `PromiseRecord` or `ERef` imports did **not** result in `any` types where they are referenced in the same file.

File: packages/marshal/src/helpers/internal-types.js
Comment (erights) [reply packages/marshal/src/helpers/internal-types.js]: Thanks! Done.

File: packages/marshal/src/helpers/remotable.js
Comment (erights) [reply packages/marshal/src/helpers/remotable.js]: Done

File: packages/vats/src/ibc.js
Comment (erights) [reply packages/vats/src/ibc.js:30]: Done!

I synced my vscode to my workspace typescript version as you suggest at https://github.com/Agoric/agoric-sdk/issues/3840#issuecomment-922374493

I then hovered over at least one `PromiseRecord` in every *.js file containing this identifier, and they all looked good. I also hovered over a random sampling of the occurrences of `ERef`. Both look exactly right.

File: packages/zoe/src/contractFacet/internal-types.js
Comment (katelynsills) [inline packages/zoe/src/contractFacet/internal-types.js:24]: I'm curious - what's the benefit of this change?

Comment (erights) [reply packages/zoe/src/contractFacet/internal-types.js:24]: Once I turn on `@ts-check`, the `Object=` form provokes a ts complaint on hover in vscode:

image

Comment (katelynsills) [reply packages/zoe/src/contractFacet/internal-types.js:24]: Interesting! Thanks for the explanation!

**Pull request merged**
